server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -747,6 +756,13 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
         isRemoteStoreEnabled = settings.getAsBoolean(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, false);
         remoteStoreRepository = settings.get(IndexMetadata.SETTING_REMOTE_STORE_REPOSITORY);
         isRemoteTranslogStoreEnabled = settings.getAsBoolean(IndexMetadata.SETTING_REMOTE_TRANSLOG_STORE_ENABLED, false);
+        isRemoteSnapshot = IndexModule.Type.REMOTE_SNAPSHOT.match(this);
```

Comment:
I know this works now, but leaking the `this` pointer from a Java constructor is usually a really bad idea. `IndexModule.Type.REMOTE_SNAPSHOT.match` gets a reference to a partially constructed object, which can lead to nasty failures (e.g. final fields can be observed as being unset). I'd refactor this to get the boolean value you want from the `settings` variable.

Reply:
Updated - I was trying to avoid repeating the logic under [IndexModule.match](https://github.com/opensearch-project/OpenSearch/blob/ea675c34b34d3af0efdd650861648a4ef7275881/server/src/main/java/org/opensearch/index/IndexModule.java#L465-L468) 😄

Reply:
Feel free to make a new `IndexModule.Type#match(Settings)` overloaded version